common/lib/transport/connectionmanager.js

```diff
@@ -406,6 +409,14 @@ var ConnectionManager = (function() {
 				this.persistConnection();
 			this.msgSerial = 0;
 		}
+		if(clientId) {
+			if(this.realtime.clientId && this.realtime.clientId != clientId) {
+				/* Should never happen in normal circumstances as realtime should
+				 * return an error code in case of mismatch */
+				throw new Error('Unexpected mismatch between expected and received clientId');
```

Comment:
This doesn't make sense because you're just throwing an exception back into the handler for the event that triggered this transport activation. Just log the error and take no further action.